spark2.4/src/main/scala/KRFLogScanner.scala

```diff
@@ -0,0 +1,83 @@
+import java.nio.file.{Files, Paths}
+
+import scala.collection.JavaConverters._
+
+object KRFLogScanner {
+  def main(args: Array[String]) {
+    if( args.length < 1 ) {
+      throw new Exception( "Missing first arg (required): path to log file" )
+    }
+    
+    val logFile = args(0)
+    val startDt = args(1)
+    
+    val lines = Files.lines( Paths.get(logFile) )
+    val lineItrs = lines.iterator().asScala.duplicate
+    
+    // <...> KRF.call p 4 t 470EF8CA332F3DDFD5FB00EC0DE88D9D71FA92DD6041CF82C4C35066B3ED45DF-19924525430016 expected 11999
+    //val expected = process(lineItrs._1, startDt, " expected ")
+    
+    // <...> KRF.call p 19 t 470EF8CA332F3DDFD5FB00EC0DE88D9D71FA92DD6041CF82C4C35066B3ED45DF-19924525430016 records 100
+    val actual = process(lineItrs._2, startDt, " records ")
+    
+//    println(s"\n$expected\tIn Kafka")
+//    println(s"$actual\tRetrieved")
+
+//    val splits = lines.iterator().asScala
+//      .filter( ln => ln.compareTo(startDt) > 0 )
+//      .filter( ln => 
+//        ln.contains("KRF.call p") && ln.contains(" records ")
+//      ).map( ln => ln.split(" ") )
+//      .duplicate
+//      
+//    val recordCount = splits._1
+//      .map( lnry => lnry(lnry.length - 1).toLong )
+//      .sum
+//    
+//    println(recordCount)
+//    
+//    val partitions = splits._2
+//      .map( lnry => lnry(lnry.length - 3) )
+//      .toSeq
+//      .distinct
+//      .sorted
+//      .mkString
+//    
+//    println(partitions)
```

Comment:
Please remove commented code, or, if it's there for a purpose, add a comment describing it, e.g. 'uncomment this code if you want to xxx...'